### PR TITLE
Autocomplete for cat, ln, ls, pwd, shutdown

### PR DESCRIPTION
**Pull request by drew-gross**

### CODE REVIEW

**Comment by RobertoUa** [comment]
Wow! We now have man-pages parsing. Amazing.
Although I think it shouldn't work like this. BS should parse them automatically (user writes a command, BS checks if it has a man page, if not - don't check again for another like week). And probably a shell command to rebuild completitions from history or just remove saved data.

**Comment by shockone** [comment]
@RobertoUa, yeah, it's a good optimization, and it's what Fish does too. But it's still an optimization, and we can worry about it later.

File: src/utils/ManPages.ts [hunk 1]
```diff
@@ -18,6 +17,14 @@ const combineManPageLines = (lines: string[]) => lines
...
 // Note: this is still pretty experimental. If you want to do man page parsing
 // for a new command, expect to have to make some changes here.
+
+// TODO: Handle option descriptions that have empty lines. Unblocks:
+// -p and -R in cp
+// df
+// locate
+// TODO: Handle nested options. Unblocks:
+// dd
+
```
Comment (shockone) [inline src/utils/ManPages.ts]: I think we should start writing tests for this component.

**Comment by RobertoUa** [comment]
You write auto parsing once instead of adding manual parsing only for some commands.

**Comment by shockone** [comment]
Ah, I see now. We can pretty easily add a man autocompletion provider to all executables with one line of code, but also maybe a hand-written provider should supersede it, because it's supposedly of better quality. What do you think?

**Comment by RobertoUa** [comment]
Sure, BS should support different levels of autocompletitions with defined priority.

**Comment by drew-gross** [comment]
@RobertoUa currently the man page parser will fail on most man pages as it can only handle basic stuff. That's why I've only enabled it for commands where I've checked that it works. I agree that hand-written providers should supersede man pages, possibly even on an option-by-option level. The man pages parse is still really in early stages, I'm sure we'll make it work better as we go.

@shockone I'll start adding tests in the next PR.

**Pull request merged**